@smaticka, thanks for creating it. Coding norms are failing with:
```
./balloon_sonde_json2ioda.py:157:9: E266 too many leading '#' for block comment
```

--- a/src/conventional/balloon_sonde_json2ioda.py
+++ b/src/conventional/balloon_sonde_json2ioda.py
@@ -153,10 +153,19 @@ def main(args):
         latitude = [file['observations'][ii]['latitude'] for ii in range(len(file['observations']))]
         longitude = [file['observations'][ii]['longitude'] for ii in range(len(file['observations']))]
         stationIdentification = [file['observations'][ii]['mission_name'] for ii in range(len(file['observations']))]
-        pressure = [file['observations'][ii]['pressure'] for ii in range(len(file['observations']))]
-        # NOTE: THESE 2 ARE TEMPORARILY REVERSED UNTIL DATA LABELING CONFIRMED
-        windEastward = [file['observations'][ii]['speed_y'] for ii in range(len(file['observations']))]
-        windNorthward = [file['observations'][ii]['speed_x'] for ii in range(len(file['observations']))]
+        pressure = [file['observations'][ii]['pressure'] for ii in range(len(file['observations']))] 
+        # Check when the file was created to handle wind components accordingly
+        dat_created = datetime.strptime(file['CreateDateTime'],"%Y-%m-%dT%H:%M:%S.%f")
+        # UPDATE THIS DATE WHEN Windborne CHANGES THEIR OBS RECORDS
+        # CreateDateTime description is not reported on their website. It appears to be either the 
+        # uploaded or downloaded date
+        dat_updated = datetime.strptime('2024-12-31T00:00:00',"%Y-%m-%dT%H:%M:%S")
+        if dat_created<dat_updated:
+          windEastward = [file['observations'][ii]['speed_y'] for ii in range(len(file['observations']))]
+          windNorthward = [file['observations'][ii]['speed_x'] for ii in range(len(file['observations']))]
+        else:
+          windEastward = [file['observations'][ii]['speed_x'] for ii in range(len(file['observations']))]
+          windNorthward = [file['observations'][ii]['speed_y'] for ii in range(len(file['observations']))]
         #
         airTemperature = [file['observations'][ii]['temperature'] for ii in range(len(file['observations']))]
         dateTime = [file['observations'][ii]['timestamp'] for ii in range(len(file['observations']))]  # datetime

--- a/src/conventional/balloon_sonde_json2ioda.py
+++ b/src/conventional/balloon_sonde_json2ioda.py
@@ -154,10 +154,10 @@ def main(args):
         longitude = [file['observations'][ii]['longitude'] for ii in range(len(file['observations']))]
         stationIdentification = [file['observations'][ii]['mission_name'] for ii in range(len(file['observations']))]
         pressure = [file['observations'][ii]['pressure'] for ii in range(len(file['observations']))]
-        #### NOTE: THESE 2 ARE TEMPORARILY REVERSED UNTIL DATA LABELING CONFIRMED
+        # NOTE: THESE 2 ARE TEMPORARILY REVERSED UNTIL DATA LABELING CONFIRMED
         windEastward = [file['observations'][ii]['speed_y'] for ii in range(len(file['observations']))]
         windNorthward = [file['observations'][ii]['speed_x'] for ii in range(len(file['observations']))]
-        ####
+        #
         airTemperature = [file['observations'][ii]['temperature'] for ii in range(len(file['observations']))]
         dateTime = [file['observations'][ii]['timestamp'] for ii in range(len(file['observations']))]  # datetime
 

--- a/src/conventional/balloon_sonde_json2ioda.py
+++ b/src/conventional/balloon_sonde_json2ioda.py
@@ -153,19 +153,19 @@ def main(args):
         latitude = [file['observations'][ii]['latitude'] for ii in range(len(file['observations']))]
         longitude = [file['observations'][ii]['longitude'] for ii in range(len(file['observations']))]
         stationIdentification = [file['observations'][ii]['mission_name'] for ii in range(len(file['observations']))]
-        pressure = [file['observations'][ii]['pressure'] for ii in range(len(file['observations']))] 
+        pressure = [file['observations'][ii]['pressure'] for ii in range(len(file['observations']))]
         # Check when the file was created to handle wind components accordingly
-        dat_created = datetime.strptime(file['CreateDateTime'],"%Y-%m-%dT%H:%M:%S.%f")
+        dat_created = datetime.strptime(file['CreateDateTime'], "%Y-%m-%dT%H:%M:%S.%f")
         # UPDATE THIS DATE WHEN Windborne CHANGES THEIR OBS RECORDS
-        # CreateDateTime description is not reported on their website. It appears to be either the 
+        # CreateDateTime description is not reported on their website. It appears to be either the
         # uploaded or downloaded date
-        dat_updated = datetime.strptime('2024-12-31T00:00:00',"%Y-%m-%dT%H:%M:%S")
-        if dat_created<dat_updated:
-          windEastward = [file['observations'][ii]['speed_y'] for ii in range(len(file['observations']))]
-          windNorthward = [file['observations'][ii]['speed_x'] for ii in range(len(file['observations']))]
+        dat_updated = datetime.strptime('2024-12-31T00:00:00', "%Y-%m-%dT%H:%M:%S")
+        if dat_created < dat_updated:
+            windEastward = [file['observations'][ii]['speed_y'] for ii in range(len(file['observations']))]
+            windNorthward = [file['observations'][ii]['speed_x'] for ii in range(len(file['observations']))]
         else:
-          windEastward = [file['observations'][ii]['speed_x'] for ii in range(len(file['observations']))]
-          windNorthward = [file['observations'][ii]['speed_y'] for ii in range(len(file['observations']))]
+            windEastward = [file['observations'][ii]['speed_x'] for ii in range(len(file['observations']))]
+            windNorthward = [file['observations'][ii]['speed_y'] for ii in range(len(file['observations']))]
         #
         airTemperature = [file['observations'][ii]['temperature'] for ii in range(len(file['observations']))]
         dateTime = [file['observations'][ii]['timestamp'] for ii in range(len(file['observations']))]  # datetime

